openassessment/fileupload/api.py

```diff
@@ -251,16 +336,19 @@ def create_shared_upload(self, fileupload):
             logger.error("Unable to create shared upload. " + str(e))
             raise e
 
-    def delete_upload(self, index):
+    def get_file_key(self, index):
+        return get_student_file_key(self.block.get_student_item_dict(), index)
+
+    def delete_upload(self, index, delete_shared_upload=False):
         """
         Given a file index to remove, null out its metadata in our stored file metadata fields
 
         Args:
-            index: file index to remove
-
-        Returns: newly updated FileUpload records.
+            index (integer): file index to remove
+            delete_shared_upload (boolean): If true, will also cause any related ``SharedFileUpload`` records
+                to be deleted. Defaults to False.
```

Comment:
Why would the default be false? We're calling the API to actually delete the file here no matter what, why would we ever want to keep the SharedFileUpload pointing at nothing?

Reply:
👍